src/lib.rs

```diff
@@ -1,4 +1,5 @@
-#![no_std]
+#[cfg(not(feature="std"))]
```

Comment:
Also, I am interested why you made this change. I have seen a few people use this `cfg` macro approach to specify `no_std`, but I never understood the exact reason. It seems more idiomatic, and that reason alone is enough for me to be in favor of this change, but I'd love to learn if there are other reasons why one way is better than the other.

Reply:
sorry fixed it, it should've been 
```
#![cfg_attr(not(feature = "std"), no_std)]
```

It's a pattern i've been using and learned while building on Substrate. It allows us to switch off feature flags I think, so that it defaults to no_std but that it can compile with std features if there's ever any new features to support that can't be written in no-std.

Reply:
Ah interesting. Well that's good to know. Thanks!